client/my-sites/people/people-section-nav/index.jsx

```diff
@@ -51,63 +59,66 @@ let PeopleNavTabs = React.createClass( {
 			</NavTabs>
 		);
 	}
-} );
+}
 
-module.exports = React.createClass( {
+class PeopleSectionNav extends React.PureComponent {
+	constructor( props ) {
+		super( props );
+	}
 
-	displayName: 'PeopleSectionNav',
+	canSearch() {
+		const { site, filter } = this.props;
 
-	canSearch: function() {
 		// Disable search for wpcom followers and viewers
-		if ( this.props.filter ) {
-			if ( 'followers' === this.props.filter || 'viewers' === this.props.filter ) {
+		if ( filter ) {
+			if ( 'followers' === filter || 'viewers' === filter ) {
 				return false;
 			}
 		}
 
-		if ( ! this.props.site.jetpack ) {
+		if ( ! site.jetpack ) {
 			// wpcom sites will always support search
 			return true;
 		}
-
-		if ( 'team' === this.props.filter && ! this.props.site.versionCompare( '3.7.0-beta', '>=' ) ) {
+		if ( 'team' === filter && versionCompare( site.options.jetpack_version, '3.7.0-beta' ) < 0 ) {
 			// Jetpack sites can only search team on versions of 3.7.0-beta or later
 			return false;
 		}
 
 		return true;
-	},
+	}
 
-	getFilters: function() {
-		var siteFilter = this.props.site.slug,
+	getFilters() {
+		const { site } = this.props,
+			siteFilter = ( site && site.slug ) ? site.slug : '',
```

Comment:
Alternatively if we only need a a few props, we can pass exactly that through. For example:
```
import { getSiteSlug, getSelectedSiteId, isJetpackSite } from 'client/state/sites/selectors';
```
render() {
   const { siteSlug: siteFilter } = this.props;
//...
}
//...
mapStateToProps( state ) {
    const selectedSiteId = getSelectedSiteId( state );
    return {
        siteSlug: getSiteSlug( state, siteId ),
        isJetpack: isJetpackSite( state, siteId )
    }
}

```